junit/src/main/java/org/kaazing/k3po/junit/rules/ScriptRunner.java

```diff
@@ -137,12 +139,19 @@ public ScriptPair call() throws Exception {
                     case FINISHED:
                         FinishedEvent finished = (FinishedEvent) event;
                         // note: observed script is possibly incomplete
-                        String observedScript = finished.getScript();
-                        return new ScriptPair(expectedScript, observedScript);
+                        observedScript = finished.getScript();
+                        
+                        // we keep the result and read any other event is still in the buffer. We will return after the first timeout
+//                        return new ScriptPair(expectedScript, observedScript);
+                        finishedReceived = true;
+                        break;
                     default:
                         throw new IllegalArgumentException("Unrecognized event kind: " + event.getKind());
                     }
                 } catch (SocketTimeoutException e) {
+                    if (finishedReceived)
```

Comment:
The problem is that we can still receive events from the driver after the FINISHED events, no matter what is the state. In order to have the correct response DISPOSED when we send a DISPOSE, we need to clear the buffer of any other event still pending. This is why I delay returning the result until the timeout (buffer clear).

Other option is to have another loop in ScriptRunner.dispose() method to read until the DISPOSED event. This again will block the main thread waiting until a timeout occurs or the DISPOSED event is received.

Or just ignore and not read the DISPOSED event.